tracer/src/Datadog.Trace/AppSec/Waf/IResult.cs

```diff
@@ -25,7 +24,7 @@ internal interface IResult
 
         public Dictionary<string, object?>? SendStackInfo { get; }
 
-        IReadOnlyCollection<object>? Data { get; }
+        List<object>? Data { get; }
```

Comment:
I made this `IReadOnlyCollection` as I we store this structure in a way that could potentially be accessed by several threads. If you're changing it `List` do you feel confident that we have the right locking in place so it won't be accessed by several threads.

See original discussion: https://github.com/DataDog/dd-trace-dotnet/pull/5055#discussion_r1452504958

Reply:
Actually, it can be done without changing that. I have updated the code. Thank you!